syntax = "proto3";

enum MessageType {
  MOVE = 0;
  TURN_KEY = 1;
}

message Move {
  int64 id = 1;
  float xDelta = 2;
  float yDelta = 3;
  float zDelta = 4;
}

message TurnKey {
  int64 id = 1;
  bool on = 2;
}

message ApiRequest {
  MessageType type = 1;
  oneof payload {
    Move move = 2;
    TurnKey turnKey = 3;
  }
}


